mqtt3/1.0/modules/ROOT/edits/mqtt3-connector-client-id.adoc

```diff
@@ -0,0 +1,355 @@
+= Set a Client ID with MQTT Connector
+
+Client IDs are important for MQTT because they identify a connection to an MQTT broker. Moreover, they identify a specific device or client,
+and it can be useful in terms of traceability to set a client ID that clearly identifies a device's connection to the broker.
+
+MQTT brokers usually implement a mechanism that closes an old connection if a new connection request with the same client ID as the existing (old) connection is received and then accepts the new connection. MQTT brokers employ this mechanism to prevent situations such as TCP connections reaching a half-open or invalid state due to a crash in one of the communication ends, which is undesirable. The broker assumes the new incoming connection to be valid, closes the older connection, and continues using the newly established connection.
+
+The client takeover mechanism can become a problem if you want to deploy an application in a cluster, and still provide an explicit client ID that identifies the client. Anypoint Connector for MQTT (MQTT Connector) provides client ID generators that help you set a static client ID, set a client ID with a dynamic component to adjust to different deployment scenarios, or define your custom client ID generator.
+
+== Configure the Client ID Custom Expression Generator
+
+This generator enables you to set a static string prefix and an expression as a suffix. Both parameters are optional but you must provide at least one of them in order to generate a client ID. If you do not provide any parameter, an `INVALID_CLIENT_ID` exception is thrown.
+
+In the following example, the configuration takes the client ID string and appends the current date, that generates a client ID similar to `principality-123-2020-09-03`:
+
+. In Studio, navigate to the *Global Elements* tab.
+. Click *Create*.
+. In the filter box, type `mqtt` and select *MQTT3 Config*.
+. Click *OK*.
+. In the *MQTT3 Config* window, for *Connection*, select *MQTT3 URL Connection*.
+. For *Client id generator*, select *Client id custom expression generator*.
+. Set the following fields:
++
+* *Client ID*: `principality-123-`
+* *Custom Client ID Suffix*: `#[%dw 2.0 output text/plain --- now() as String {format: 'y-MM-dd'}]"`
+* *Username*: `mosquitto`
+* *Password*: `password`
+* *URL*: `tcp://broker.hivemq.com:1883`
+[start=7]
+. Click *OK*.
+
+image::mqtt3-expression-generator.png[Client ID Custom Expression Generator configuration in Global Element Properties window]
+
+In the Configuration XML editor, the `<mqtt3:client-id-custom-expression-generator>` and `customClientIdSuffix` configurations look like these:
```

Comment:
```suggestion
In the Configuration XML editor, the `<mqtt3:client-id-custom-expression-generator>` and `customClientIdSuffix` configurations look like this:
```
...because what follows is a single depiction of the two things. Let me know if this is confusing.